enterprise/server/workflow/service/service.go

```diff
@@ -1153,6 +1157,7 @@ func (ws *workflowService) createActionForWorkflow(ctx context.Context, wf *tabl
 			"--workflow_id=" + wf.WorkflowID,
 			"--trigger_event=" + wd.EventName,
 			"--bazel_command=" + ws.ciRunnerBazelCommand(),
+			"--env_overrides=" + string(envJson),
```

Comment:
we've broken things in the past when adding new flags to the ci_runner, because Go crashes when it sees unrecognized flags. There will be older ci_runners that are assigned to this task because we distribute the CI runner with the executor rather than the app, and there are people that run self-hosted workflow executors. (This also will temporarily be broken while we do the rollout)

I think ideally we'd distribute the CI runner binary with the app (rather than distributing it w/ the executors) and upload it to CAS and make it part of the Action, so that the flags are kept in lock-step with the apps. But the last time we looked at doing that we decided it was complicated, since it makes the apps depend on both macos and linux builds of the CI runner.

For now, we can conditionally pass the `--env_overrides` flag  (i.e. if env is empty then don't set the flag). (Maybe also add a comment here for future readers if you don't mind :pray: )

Reply:
Done